### PR TITLE
add sort param to splitByGroups

### PR DESCRIPTION
**Pull request by abmantis**
This is needed for https://github.com/home-assistant/home-assistant-polymer/pull/443

Should I bump the version and update the dependency version on home-assistent-polymer?

### CODE REVIEW

File: lib/group.js [hunk 1]
```diff
@@ -15,9 +15,9 @@ export function getGroupEntities(entities, group) {
 }
 
 // Split a collection into a list of groups and a 'rest' list of ungrouped
-// entities. Groups will be ordered by their order.
+// entities. Groups will be ordered by their order if 'sort' is true.
 // Returns { groups: [], ungrouped: {} }
-export function splitByGroups(entities) {
+export function splitByGroups(entities, sort = true) {
```
Comment (balloob) [inline lib/group.js]: Let's keep this function simple and remove sorting the groups array completely.

Where it is needed, the ordering can be done by the code calling this.

Make sure to update the test too.

**Comment by balloob** [comment]
Shit, I just realized that this is not going to work either.

See, now the groups are not ordered at all. Including not being ordered by the view. The problem is that in the polymer-side of things, we pass down an object of entities and then have it use this function to figure out what to do.

However, objects in JavaScript are iterated over in random order instead of insertion order.

**Comment by balloob** [comment]
So I guess that we can close this PR and solve this completely in polymer. Instead of passing down `isView`, we should pass down the order of groups to use (based on the entity ids in the view)

**Comment by balloob** [comment]
Closing this as it is no longer needed

**Comment by abmantis** [comment]
I think we should also merge this. There is no need to keep the sorting here since it is done outside.

**Comment by balloob** [comment]
Good point. Can you move the sorting into the tests? There is no guarantee that this order is going to be always like this as it's based on `Object.keys`.

**Comment by balloob** [comment]
Woohoo Thanks 🐬

**Pull request merged**